Bottom: 480bad01b6482168bf96ca40c051b2eedef70f3c
Top:    65773dc4199a5b0422b1b6b9d4f50456c4f83e7f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-02-10 20:29:46 +0000

Workaround crash where contour is passed as NULL to poly_PreContour













---

diff --git a/src/polygon.c b/src/polygon.c
index c0aa76d..181a6c3 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -247,6 +247,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
